### PR TITLE
Changed Header Guards to Pragma Once

### PR DESCRIPTION
**Pull request by Jim-Marsden**
Thanks a lot for making a contribution to SFML! 🙂

Before you create the pull request, we ask you to check the follow boxes. (For small changes not everything needs to ticked, but the more the better!)

* [x] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before? #2305 and discussed to death on discord.
* [x] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [ ] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

----

## Description

Please describe your pull request.

Replacing header guards with pragma once

## Tasks

* [x] Tested on Linux
* [x] Tested on Windows
* [x] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

## How to test this PR?
The functionality should exactly the same?

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
Screenshot 2022-12-20 at 12 03 22 PM

If you regex search for `SFML_.*_HPP` you'll find more.

**Comment by Jim-Marsden** [comment]
> #ifndef SFML_.*_HPP

Thanks, my regex is weak!

File: src/SFML/Window/OSX/SFOpenGLView.h [hunk 1]
```diff
@@ -71,8 +71,7 @@ class WindowImplCocoa;
 /// the cursor (that was disconnected from the system).
 ///
 ////////////////////////////////////////////////////////////
-@interface SFOpenGLView : NSOpenGLView
-{
+@interface SFOpenGLView : NSOpenGLView {
```
Comment (ChrisThrasher) [inline src/SFML/Window/OSX/SFOpenGLView.h:74]: I think you somehow accidentally used clang-format-15 to format this which sadly doesn't work. If you used the `format` target to do the formatting, it should have detected that you were using v15 and prevented it from running.

Comment (Jim-Marsden) [reply src/SFML/Window/OSX/SFOpenGLView.h:74]: As mentioned in private it was that I forced CLion to do formatting.

File: include/SFML/Graphics/Transform.hpp [hunk 2]
```diff
@@ -350,7 +349,7 @@ constexpr Vector2f operator*(const Transform& left, const Vector2f& right);
 } // namespace sf
 
 
-#endif // SFML_TRANSFORM_HPP
+
```
Comment (Radix-Comet) [inline include/SFML/Graphics/Transform.hpp:353]: This is an abnormal amount of spaces for the end of the file. It doesn't hurt anything and I don't think it's a huge issue, plenty of these headers had weird spacing, but this one is like 5 where the typical is 2 or 3.

**Review by Radix-Comet — CHANGES_REQUESTED** [review]
Everything looks good except for a spacing issue in ``include/SFML/Graphics/Transform.hpp``

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

File: include/SFML/Graphics/Transform.hpp [hunk 3]
```diff
@@ -350,7 +349,7 @@ constexpr Vector2f operator*(const Transform& left, const Vector2f& right);
...
```
Comment (kimci86) [inline include/SFML/Graphics/Transform.hpp:352]: You probably want to keep only 2 blank lines here like it is done in other files.

**Review by kimci86 — COMMENTED** [review]
Looks good to me expect for one file. See my comment below.

File: include/SFML/Graphics/Transform.hpp (hunk 3 above)
Comment (kimci86) [reply include/SFML/Graphics/Transform.hpp:352]: Oh I was ninja'd by @Radix-Comet :smile:

Comment (Jim-Marsden) [reply include/SFML/Graphics/Transform.hpp:352]: Thank you both @kimci86 @Radix-Comet; that was a lot to go through!

I'll address it soon.

**Comment by kimci86** [comment]
Header guard are remaining in those files:
- `tools/xcode/templates/SFML/SFML App.xctemplate/ResourcePath.hpp`
- `examples/shader/Effect.hpp`

**Comment by Jim-Marsden** [comment]
> * tools/xcode/templates/SFML/SFML App.xctemplate/ResourcePath.hpp

How did you find those? Any chance there are more I missed?

**Comment by kimci86** [comment]
> How did you find those? Any chance there are more I missed?

I did a search for "#ifndef" in VSCodium. Searching in files matching `*.hpp` pattern, excluding `extlibs,gl.h,vulkan.h`.
I don't see any other match.

**Comment by Jim-Marsden** [comment]
> > How did you find those? Any chance there are more I missed?
> 
> I did a search for "#ifndef" in VSCodium. Searching in files matching `*.hpp` pattern, excluding `extlibs,gl.h,vulkan.h`. I don't see any other match.

Awesome, I loooked for just #ifndef and I didn't see anything else either.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by ChrisThrasher — APPROVED** [review]
Screenshot 2022-12-21 at 3 50 01 PM

We love to see it

**Review by Bambo-Borris — APPROVED** [review]
Happy to see this change, good work @Jim-Marsden 🥳

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
Thank you for tackling this! 🥳

**Pull request merged**